lib/rubocop/cop/sorbet/forbid_nested_t_must.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Sorbet
+      # Checks for nested `T.must` calls and suggests to use
+      # a single `T.must` around a safe-navigation operator chain.
+      #
+      # @example
+      #
+      #    # bad
+      #    T.must(T.must(A.b).c)
+      #
+      #    # bad
+      #    T.must(T.must(T.must(A.b&.b).b).c)
+      #
+      #    # bad
+      #    T.must(T.must(A.b)[:test])
+      #
+      #    # good
+      #    T.must(A.b&.c)
+      #
+      #    # good
+      #    T.must(A.b&.b&.b&.c)
+      #
+      #
+      #    # good
+      #    T.must(A.b&.[](:test))
+      #
+      #    # good
+      #    T.must(A.d(T.must(A.b&.c)))
+      class ForbidNestedTMust < Base
+        extend AutoCorrector
+        include IgnoredNode
+
+        MSG =
+          "Please avoid nesting `T.must` calls, instead use a " \
+            "single `T.must` around a safe-navigation operator chain"
```

Comment:
What do you think of Jez's "conditional send" naming? It's fitting here, given that he's one of the primary contributors of Sorbet :D

```suggestion
            "single `T.must` around a conditional-send (`&.`) chain"
```

Reply:
Sure! Let's use that instead! :)